recommendations/local-resource-conservation.md

```diff
@@ -0,0 +1,333 @@
+reputation score rewards fast resolving, successful payments and penalizes 
+slow resolving payments (regardless of successful resolution).
+
+We define the following parameters: 
+* `resolution_period`: the amount of time a HTLC is allowed to resolve in that 
+  classifies as "good" behavior, expressed in seconds. The recommended default 
+  is 90 seconds (given that the protocol allows for a 60 second MPP timeout). 
+* `resolution_time`: the amount of time elapsed in seconds between a HTLC being 
+  added to and removed from the local node's commitment transaction. 
+* `fees`: the fees that are offered to the local node to forward the HTLC, 
+  expressed in milli-satoshis.
+
+We define the `opportunity_cost` of the time a HTLC takes to resolve:
+`opportunity_cost`: `ceil ( (resolution_time - resolution_period) / resolution_period) * fees`
+
+Given that `resolution_time` will be > 0 in practice, `opportunity_cost` is 0 
+for HTLCs that resolve within `resolution_period`, and charges the `fees` that 
+the HTLC would have earned per period it is held thereafter. This cost accounts 
+for the slot and liquidity that could have otherwise been paid for by 
+successful, fast resolving HTLCs during the `resolution_time` the HTLC was 
+locked in the channel.
+
+Each HTLC's contribution to reputation is expressed by its `effective_fee` 
+which is determined by its endorsement, resolution time and outcome: 
+* if `endorsed` is non-zero in `update_add_htlc`:
+  * if successfully resolved with `update_fulfill_hltc`: 
+    * `effective_fees` = `fees` - `opportunity_cost` 
+  * otherwise: 
+    * `effective_fees` = -`opportunity_cost` 
+* otherwise: 
+  * if successfully resolved AND `resolution_time` <= `resolution_period`
+    * `effective_fees` = `fees`
+  * otherwise: 
+    * `effective_fees` = 0
+
+##### Rationale
+Reputation is negatively affected by slow-resolving HTLCs (regardless of whether
+they are settled or failed) to ensure that reputation scoring reacts to bad 
+behavior. HTLCs that resolve within a period of time that is considered 
+reasonable do not decrease reputation, as some rate of failure is natural in 
+the network.
+
+The resolution of HTLCs with `endorsed` set to `0` do not have a negative 
+impact on reputation because the forwarding node made no promises about their 
+behavior. They are allowed to build reputation when they resolve successfully 
+within the specified `resolution_period` to allow new nodes in the network to 
+bootstrap reputation. 
+
+#### Outgoing Link Revenue
+Outgoing link revenue measures the damage to the local node (ie, the loss in 
+forwarding fees) that a slow jamming attack can incur. For an individual link, 
+this is equal to the total revenue forwarded in both directions over the 
+maximum allowed HTLC hold time.
+
+We define the following parameters: 
+* `outgoing_link_window`: the largest cltv delta from the current block height 
```

Comment:
Do we want this to be the largest cltv delta, or do we want more flexibility with setting the window size?